### PR TITLE
Expose user language through DAV

### LINKED ISSUE #28449: Expose user language through a DAV property
Required for https://github.com/nextcloud/calendar/issues/3414

Needs to be added here
https://github.com/nextcloud/server/blob/9585c4f2561efd04a85b72b1100bd3593129b00c/apps/dav/lib/Connector/Sabre/Principal.php#L504-L519
And probably a few adjustments to make sure it's read-only.

### PR DESCRIPTION
**Pull request by tcitworld**
Introduces the `'{http://nextcloud.com/ns}language'` prop that gives the user's language

Closes #28449

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

File: apps/dav/appinfo/v1/caldav.php [hunk 1]
```diff
@@ -52,6 +52,7 @@
 	\OC::$server->query(\OCA\DAV\CalDAV\Proxy\ProxyMapper::class),
 	\OC::$server->get(KnownUserService::class),
 	\OC::$server->getConfig(),
+	\OC::$server->getL10NFactory(),
```
Comment (st3iny) [inline apps/dav/appinfo/v1/caldav.php:55]:
```suggestion
	\OC::$server->get(IFactory::class),
```

File: apps/dav/appinfo/v1/carddav.php [hunk 2]
```diff
@@ -55,6 +55,7 @@
 	\OC::$server->query(\OCA\DAV\CalDAV\Proxy\ProxyMapper::class),
 	\OC::$server->get(KnownUserService::class),
 	\OC::$server->getConfig(),
+	\OC::$server->getL10NFactory(),
```
Comment (st3iny) [inline apps/dav/appinfo/v1/carddav.php:58]:
```suggestion
	\OC::$server->get(IFactory::class),
```

File: apps/dav/lib/Command/CreateCalendar.php [hunk 3]
```diff
@@ -87,7 +87,8 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 			\OC::$server->getAppManager(),
 			\OC::$server->query(ProxyMapper::class),
 			\OC::$server->get(KnownUserService::class),
-			\OC::$server->getConfig()
+			\OC::$server->getConfig(),
+			\OC::$server->getL10NFactory(),
```
Comment (st3iny) [inline apps/dav/lib/Command/CreateCalendar.php:91]:
```suggestion
			\OC::$server->get(IFactory::class),
```

File: apps/dav/lib/RootCollection.php [hunk 4]
```diff
@@ -73,7 +73,8 @@ public function __construct() {
 			\OC::$server->getAppManager(),
 			$proxyMapper,
 			\OC::$server->get(KnownUserService::class),
-			\OC::$server->getConfig()
+			\OC::$server->getConfig(),
+			\OC::$server->getL10NFactory()
```
Comment (st3iny) [inline apps/dav/lib/RootCollection.php:77]:
```suggestion
			\OC::$server->get(IFactory::class)
```

**Review by st3iny — COMMENTED** [review]
Please replace all calls to the deprecated method getL10NFactory() and don't forget to import IFactory.

You could also use this patch I prepared:
[0001-Fix-deprecations.txt](https://github.com/nextcloud/server/files/7020260/0001-Fix-deprecations.txt)

**Comment by tcitworld** [comment]
@st3iny Since all around calls also use depreciated methods (`\OC::$server->getConfig()` and such), I planned to do this in a second step.

**Comment by st3iny** [comment]
> @st3iny Since all around calls also use depreciated methods (`\OC::$server->getConfig()` and such), I planned to do this in a second step.

Fair enough. I'm just not comfortable with adding deprecated code to the repository.

**Review by st3iny — APPROVED** [review]

**Pull request merged**